cli/internal/cmd/init.go

```diff
@@ -249,17 +259,72 @@ func (i *initCmd) writeOutput(idFile clusterid.File, resp *initproto.InitRespons
 	if err := fileHandler.Write(constants.AdminConfFilename, resp.Kubeconfig, file.OptNone); err != nil {
 		return fmt.Errorf("writing kubeconfig: %w", err)
 	}
-	i.log.Debugf("Wrote out kubeconfig")
+	i.log.Debugf("Wrote kubeconfig to file")
+	if mergeConfig {
+		if err := i.mergeKubeonfig(clusterName, fileHandler); err != nil {
+			return fmt.Errorf("merging kubeconfig: %w", err)
+		}
+		i.log.Debugf("Merged kubeconfig to existing file")
+		writeRow(tw, "Kubernetes configuration merged with default config", "")
+	}
 	idFile.OwnerID = ownerID
 	idFile.ClusterID = clusterID
 
 	if err := fileHandler.WriteJSON(constants.ClusterIDsFileName, idFile, file.OptOverwrite); err != nil {
-		return fmt.Errorf("writing Constellation id file: %w", err)
+		return fmt.Errorf("writing Constellation ID file: %w", err)
 	}
-	i.log.Debugf("Wrote out Constellation id file")
+	i.log.Debugf("Wrote out Constellation ID file")
+
+	if !mergeConfig {
+		fmt.Fprintln(wr, "You can now connect to your cluster by executing:")
+		fmt.Fprintf(wr, "\texport KUBECONFIG=\"$PWD/%s\"\n", constants.AdminConfFilename)
+	} else {
+		fmt.Fprintln(wr, "Constellation kubeconfig merged with default config.")
 
-	fmt.Fprintln(wr, "You can now connect to your cluster by executing:")
-	fmt.Fprintf(wr, "\texport KUBECONFIG=\"$PWD/%s\"\n", constants.AdminConfFilename)
+		if os.Getenv(clientcmd.RecommendedConfigPathEnvVar) != "" {
+			fmt.Fprintln(wr, "Warning: KUBECONFIG environment variable is set.")
+			fmt.Fprintln(wr, "You may need to unset it to use the default config and connect to your cluster.")
+		} else {
+			fmt.Fprintln(wr, "You can now connect to your cluster.")
+		}
+	}
+	return nil
+}
+
+func (i *initCmd) mergeKubeonfig(clusterName string, fileHandler file.Handler) error {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	loadingRules.Precedence = []string{
+		constants.AdminConfFilename,   // load our config first so it takes precedence
+		clientcmd.RecommendedHomeFile, // then load the default config
+	}
+	i.log.Debugf("Kubeconfig file loading precedence: %v", loadingRules.Precedence)
+
+	// merge and flatten the kubeconfigs
+	cfg, err := loadingRules.Load()
+	if err != nil {
+		return fmt.Errorf("loading merged kubeconfig: %w", err)
+	}
+	if err := clientcmdapi.FlattenConfig(cfg); err != nil {
+		return fmt.Errorf("flattening merged kubeconfig: %w", err)
+	}
+
+	// Set the current context to the cluster we just created
+	cfg.CurrentContext = fmt.Sprintf("%s-admin@%s", clusterName, clusterName)
```

Comment:
This feels error prone to me, if we change context name in future we need to remember to change it here as well. Could we easily read out the context name instead of regenerating it?